### DISCUSSION TITLE
Customizing special keys and general docs questions

### DISCUSSION THREAD

**Discussion by asmodeus812** [ggandor/leap.nvim#186] [Q&A]
Hi i am having a very difficult time understanding a lot of the docs for this plugin in particular, there are a lot of terms being used, which are internal and specific for the plugin, like phases, and targets, etc. One such example, next and prev target. See below 

```lua
      leap.setup(vim.tbl_deep_extend("force", {
        max_phase_one_targets = 2,
        special_keys = {
            next_target = "<tab>",
            prev_target = "<s-tab>",
            next_group = "<space>",
            prev_group = "<backspace>",
            multi_accept = nil,
            multi_revert = nil
        },
    }, config._config or {}))
```

- **next/prev target** - i imagine, when you put down one or two characters you can use that binding to move the cursor to the next or prev label forward or backward (as the docs say similar to how , or ; work) without closing the leap motion, **what happens is when i press tab, it moves to the next occurrence and closes leap,** if i press shift tab, **leap is simply closed, cursor does not move**

- **multi_accept/revert** - i have no idea, what multi accept or revert is tried to understand the docs, read through them multiple times and i still have no clue what it tries to say.

- **other pet peeves** - such as safe_labels max_phase_one_targets - which i pretty much tried working out on trial error basics, as i did not want to see labels on the first key input but only after the second one as it is quite confusing **_seeing labels you can't jump to, it sometimes makes you think you might have entered two chars where you have only pressed one_**, there is no visible indication on that - ala :h showcmd

**Comment by ggandor** [comment]
> next/prev target

https://github.com/ggandor/leap.nvim/blob/5efe985cf68fac3b6a6dfe7a75fbfaca8db2af9c/doc/leap.txt#L75

> as the docs say similar to how , or ; work

No, that refers to the special function `add_repeat_mappings` (https://github.com/ggandor/leap.nvim#mappings). `prev_target` reverts the effect of `next_target`, pressing it will simply exit leap, without moving to the next occurrence first.

> multi_accept/revert

https://github.com/ggandor/leap.nvim/blob/5efe985cf68fac3b6a6dfe7a75fbfaca8db2af9c/doc/leap.txt#L466

The `multi_accept/revert` keys are only relevant for custom functions that use the `multiselect` flag. See an example in the readme.

> other pet peeves - such as safe_labels

Is there something unclear in the help entries? I really don't know what else to add...
